stored_requests/backends/db_fetcher/fetcher.go

```diff
@@ -93,6 +93,10 @@ func (fetcher *dbFetcher) FetchRequests(ctx context.Context, requestIDs []string
 	return storedRequestData, storedImpData, errs
 }
 
+func (fetcher *dbFetcher) FetchCategories() map[string]map[string]json.RawMessage {
```

Comment:
You can probably just have this return as `map[string]map[string]string` The other fetchers return `json.RawData` as their return values will be used in JSON merges, so need be transformed into `json.RawMessage`.

Actually if you want to do this by extending the current fetcher machinery, you should have it fetch just the needed categories once fed the adserver, account ID, and a list of IAB categories that need to be translated. This would match the usage of `FetchRequests()` and allow better support of the more dynamic backends when we get around to supporting them. For any category mapping that is not entirely static, you must return a static copy of the mapping to avoid a race condition where you update the copy of the map while a goroutine is busy doing translations based on that map. Much more efficient just to produce a copy of the items you actually need than copy the whole structure.

Reply:
@hhhjort Thank you for the comments! Code is now refactored: NewFetcher returns directory data and FetchCategories() now return Categories struct with unmarshalled data that is ready for usage in endpoints. Current implementation loads category data at server startup, same as stored requests, if we load it from file system. Can you please re-review changes?